src/components/Map/index.js

```diff
@@ -77,11 +103,38 @@ const MapView = ({coordinates = null, zoomLevel = DEFAULT_ZOOM}) => {
     if (!coordinates || coordinates.length !== 2) return;
     if (!coordinates.filter(value => isFinite(value)).length === 2) return;
     map.setView(coordinates, zoomLevel);
-  });
+  }, [coordinates]);
   // rendering is done by leaflet, so just return null
   return null;
 };
 
+const SearchControl = props => {
+  const map = useMap();
+  const onMarkerSet = props.onMarkerSet;
+  const intl = useIntl();
+
+  const setMarker = result => {
+    if (result.location) {
+      onMarkerSet([result.location.y, result.location.x]);
+    }
+  };
+
+  useEffect(() => {
+    const searchControl = new GeoSearchControl({
+      provider: props.provider,
+      style: 'button',
+      ...props,
+    });
+    searchControl.button.setAttribute('aria-label', intl.formatMessage(searchButtonAreaLabel));
+
+    map.addControl(searchControl);
+    return () => map.removeControl(searchControl);
+  }, []);
+
+  map.on('geosearch/showlocation', setMarker);
```

Comment:
shouldn't this go in the `useEffect` to bind the event, and consequently in the cleanup function -> unbind the event?